lib/path_manager.c

```diff
@@ -39,6 +42,33 @@ static bool is_pm_ready(struct mptcpd_pm const *pm, char const *fname)
 
 // --------------------------------------------------------------------
 
+bool mptcpd_pm_register_ops(struct mptcpd_pm *pm,
+                            struct mptcpd_pm_ops const *ops,
+                            void *user_data)
+{
+        if (pm == NULL || ops == NULL)
+                return false;
+
+        if (ops->ready        == NULL
+            && ops->not_ready == NULL) {
+                l_error("No path manager event tracking "
+                        "ops were set.");
+
+                return false;
+        }
+
+        struct pm_ops_info *const info = l_malloc(sizeof(*info));
+        info->ops = ops;
+        info->user_data = user_data;
+
+        bool const registered = l_queue_push_tail(pm->event_ops, info);
+
+        if (!registered)
+                l_free(info);
+
+        return registered;
```

Comment:
As is, the user's `ready` callback will not be called if the PM is ready prior to `ops` registration.  Should we schedule/trigger a call to the user-provided `ready` callback in that case?

Reply:
If the caller of `mptcpd_pm_register_ops()` can also make a synchronous call to see if the PM is ready, I don't think it's necessary to schedule a callback.

I think of these callbacks as firing when something changes (edge-triggered). Since the caller is already "awake" to call this register function, it can also check the current state of the PM and proceed accordingly.